src/easier68k/core/models/list_file.py

```diff
@@ -0,0 +1,157 @@
+import json
+import re
+
+"""
+List File
+
+Represents the output from the assembler that contains all of the instructions and where in the
+destination memory they should end up.
+"""
+MAX_MEMORY_LOCATION = 16777216  # 2^24
+
+class ListFile:
+    """
+    Represents assembled instructions and their locations in memory
+    """
+    def __init__(self):
+        """
+        Constructor
+        """
+        # the keys of data must be strings so that they will work with JSON
+        # but when working with it, integers are a lot cleaner and make more sense
+        # so all of the interfaces that work with it are going to use ints
+        # but internally it will use strings
+        self.data = {}
+        self.symbols = {}
+
+    def insert_data(self, location: int, data: str):
+        """
+        Inserts the data at the given location into the list file
+        This data should be a string of hexadecimal data
+        :param location:
+        :param data:
+        :return:
+        """
+        assert location >= 0, 'Location is out of bounds!'

```

Comment:
Minor nitpick, this could be combined into a single assert statement (`0 <= location < MAX_MEMORY_LOCATION`)

Reply:
True, but we could mention that when the location is too large it is beyond the range of possible values for location.